inc/packages/class-upgrader.php

```diff
@@ -251,20 +304,32 @@ protected function run_install( string $destination, $options ) {
 		return $this->on_run_complete( $result, $options );
 	}
 
+	/**
+	 * Set trusted keys.
+	 *
+	 * @return array
+	 */
 	protected function set_trusted_keys() {
 		$doc = get_did_document( $this->package->id );
 		if ( is_wp_error( $doc ) ) {
 			return [];
 		}
 
-
 		$valid_keys = $doc->get_fair_signing_keys();
 
-		// todo: re-encode from multibase to base64
+		// todo: re-encode from multibase to base64.
 		// return $valid_keys;
 		return [];
 	}
 
+	/**
+	 * Install plugin.
+	 *
+	 * @param  bool $clear_cache Boolean to clear cache.
+	 * @param  bool $overwrite Boolean to overwrite.
+	 *
+	 * @return bool|WP_Error True if the installation was successful, false or a WP_Error object otherwise.
```

Comment:
`WP_Upgrader::$result` is documented as `array|WP_Error` [Ref](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/class-wp-upgrader.php#L93).

This should be:
```
* @return bool|array|WP_Error True if the installation was successful, an array or a WP_Error object otherwise.
```

Reply:
I think it's actually [here](https://github.com/WordPress/wordpress-develop/blob/99499d88a83fb7d5d0b66aa49c6b3108e6bd78df/src/wp-admin/includes/class-plugin-upgrader.php#L118).

Reply:
Strange. `WP_Upgrader::$result` should never be `false` based on the property's documentation. 🤔

Reply:
```
		if ( ! $this->result || is_wp_error( $this->result ) ) {
			return $this->result;
		}
```

Implies it could be `false`